postgres-operator-sdk/README.md

```diff
@@ -0,0 +1,288 @@
+
+Register a Postgres object using the kubectl command:
+
+```
+kubectl create -f deploy/crd.yaml
+```
+
+You should see the message we inserted in wrap.go pop up in the log file (The log terminal window). 
+
+Now it is time to begin the experiment:
+
+The way we will measure performance of the library is based on the number of API calls that appear over a period of time during experimentation. The reason for this is because at the moment there is no way to differentiate between Resync API calls and API calls resulting from Library calls in Postgrescontroller.
+
+To begin the experiment, run the Postgres Operator as defined by the steps above. 
+
+The format I followed was a 5 minute interval involving creation of DB as well as Users in the Postgrescontroller.
+
+Minute 0: Turn on Kubernetes cluster and record calls.
+
+Minute 2: Create this resource:
+
+```
+kubectl create -f deploy/initializeclient.yaml
+```
+
+Minute 3:
+```
+kubectl create -f deploy/add-db.yaml
+```
+
+Minute 4: 
+
+```
+kubectl create -f deploy/add-user.yaml
+```
+
+Minute 5: Conclude experiment.
+
+For comparison, we ran the same experiment with Postgres custom-controller based on sample-controller.
+
+
+Clone this repository [Custom Controller](https://github.com/cloud-ark/kubeplus/tree/master/postgres-crd-v2):
+
+```
+git clone https://github.com/cloud-ark/kubeplus/tree/master/postgres-crd.git
+```
+
+Follow the steps outlined in the Github repo. No major changes are required. 
+
+Perform the same experiment on the Sample Postgres Controller.
+
+
+## Results
+
+Create a table with following rows and columns:
+
+Rows: 
+- postgres based on sample-controller
+- postgres with Operator SDK without Listers
+- postgres with Operator SDK with Listers
+
+Columns:
+- Number of API calls for initializeclient
+- Number of API calls for add-db
+- Number of API calls for add-user
+
+Is it possible for you to distinguish above three calls in your log file? I think it should be possible.
+Then the way you want to fill above columns is as follows:
+- for each run of the experiment calculate calls for initializeclient and then take the average over 3 runs
+- for each run of the experiment calculate calls for add-db and then take the average over 3 runs
+- for each run of the experiment calculate calls for add-user and then take the average over 3 runs
+
+
+## Experience of using Operator SDK
+
+While running the experiment originally, I faced a challenge in dealing with circulating events. This involved events that kept circulating in the queue even after being dealt with. This added unnecessary overhead and I had to manually code in a way to ignore this event from circulating. (Relevant issues [Issue #335](https://github.com/operator-framework/operator-sdk/issues/335), [Issue 268](https://github.com/operator-framework/operator-sdk/issues/268))
+
+One way I also found to fix this is to set the resyncPeriod to 0 in the main.go file. --> Add reference where this suggestion was made
+
```

Comment:
They mention with predicate-based event filtering (https://github.com/operator-framework/operator-sdk/issues/268) it will be possible to filter out events whose resource versions are not changed. You should mention this point.